src/main/tut/1.10-natural-transformations.md

```diff
@@ -0,0 +1,244 @@
+```Haskell
+αₐ :: F a -> G a
+```
+```scala
+val αA: F[A] => G[A]
+```
+................
+```Haskell
+F f :: F a -> F b
+G f :: G a -> G b
+```
+```scala
+val Ff: F[A] => F[B]
+val Gf: G[A] => G[B]
+```
+................
+```Haskell
+alphaₐ :: F a -> G a
+```
+```scala
+val alphaA: F[A] => G[A]
+```
+................
+```Haskell
+alpha :: forall a . F a -> G a
+```
+```scala
+def alpha[A]: F[A] => G[A]
+```
+................
+```Haskell
+safeHead :: [a] -> Maybe a
+safeHead [] = Nothing
+safeHead (x:xs) = Just x
+```
+```tut:silent
+def safeHead[A]: List[A] => Option[A] = {
+  case Nil => None
+  case x :: xs => Some(x)
+}
+```
+................
+```Haskell
+fmap f . safeHead = safeHead . fmap f
+```
+```scala
+(fmap(f) compose safeHead) ==
+  (safeHead compose fmap(f))
+```
+................
+```Haskell
+fmap f (safeHead []) = fmap f Nothing = Nothing
+safeHead (fmap f []) = safeHead [] = Nothing
+```
+```scala
+fmap(f)(safeHead(List.empty)) ==
+  fmap(f)(None) == None
+safeHead(fmap(f)(List.empty)) ==
+  safeHead(List.empty) == None
+```
+................
+```Haskell
+fmap f (safeHead (x:xs)) = fmap f (Just x) = Just (f x)
+safeHead (fmap f (x:xs)) = safeHead (f x : fmap f xs) = Just (f x)
+```
+```scala
+fmap(f)(safeHead(x :: xs)) ==
+  fmap(f)(Some(x)) == Some(f(x))
+safeHead(fmap(f)(x :: xs)) ==
+  safeHead(f(1) :: fmap(f)(xs)) == Some(f(x))
+```
+................
+```Haskell
+fmap f [] = []
+fmap f (x:xs) = f x : fmap f xs
+```
+```tut:silent
+def fmap[A, B](f: A => B)
+    : List[A] => List[B] = {
+  case Nil => Nil
+  case x :: xs => f(x) :: fmap(f)(xs)
+}
+```
+................
+```Haskell
+fmap f Nothing = Nothing
+fmap f (Just x) = Just (f x)
+```
+```tut:silent
+def fmap[A, B](f: A => B)
+    : Option[A] => Option[B] = {
+  case None => None
+  case Some(x) => Some(f(x))
+}
+```
+................
+```Haskell
+length :: [a] -> Const Int a
+length [] = Const 0
+length (x:xs) = Const (1 + unConst (length xs))
+```
+```scala
+def length[A]: List[A] => Const[Int, A] = {
+  case Nil => Const(0)
+  case x :: xs => Const(1 + unConst(length(xs)))
+}
+```
+................
+```Haskell
+unConst :: Const c a -> c
+unConst (Const x) = x
+```
+```scala
+def unConst[C, A]: Const[C, A] => C = {
+  case Const(x) => x
+}
+```
+................
+```Haskell
+length :: [a] -> Int
+```
+```scala
+def length[A]: List[A] => Int
+```
+................
+```Haskell
+scam :: Const Int a -> Maybe a
+scam (Const x) = Nothing
+```
+```scala
+def scam[A]: Const[Int, A] => Option[A] = {
+  case Const(x) => None
+}
+```
+................
+```Haskell
+newtype Reader e a = Reader (e -> a)
+```
+```tut:silent
+case class Reader[E, A](run: E => A)
+```
+................
+```Haskell
+instance Functor (Reader e) where
+    fmap f (Reader g) = Reader (\x -> f (g x))
+```
+```scala
+implicit def readerFunctor[E] =
+  new Functor[({type T[A] = Reader[E, A]})#T] {
```

Comment:
Should it be prettified with kind-projector? or keep it vanilla scala?

Reply:
Added comments. Is it readable?

Reply:
LGTM. Thanks!